### ISSUE TITLE
Custom vich form issue

### ISSUE THREAD

**Issue by fluxion8** [dustin10/VichUploaderBundle#1199]
|    Q            |   A
|---------------- | ------
| Bundle version  |1.17 
| Symfony version | 5.2
| PHP version     | 7.4

Spent my time a lot this issue.

When i using custom form following ussue happening.
Argument 2 passed to Vich\UploaderBundle\Form\Type\VichFileType::resolveUriOption() must be an object, array given, called in C:\Users\onrya\PhpstormProjects\forum-symfony\vendor\vich\uploader-bundle\src\Form\Type\VichImageType.php on line 89

**Entity**

```
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\HttpFoundation\File\File;
use Vich\UploaderBundle\Mapping\Annotation as Vich;

/**
 * @ORM\Entity
 * @Vich\Uploadable
 */
class MediaImage
{
    /**
     * @ORM\Id
     * @ORM\Column(type="integer")
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    private $id;

    // ... other fields

    /**
     * NOTE: This is not a mapped field of entity metadata, just a simple property.
     *
     * @Vich\UploadableField(mapping="media_image", fileNameProperty="imageName", size="imageSize")
     *
     * @var File|null
     */
    private $imageFile;

    /**
     * @ORM\Column(type="string")
     *
     * @var string|null
     */
    private $imageName;

    /**
     * @ORM\Column(type="integer")
     *
     * @var int|null
     */
    private $imageSize;

    /**
     * @ORM\Column(type="datetime")
     *
     * @var \DateTimeInterface|null
     */
    private $updatedAt;

    /**
     * If manually uploading a file (i.e. not using Symfony Form) ensure an instance
     * of 'UploadedFile' is injected into this setter to trigger the update. If this
     * bundle's configuration parameter 'inject_on_load' is set to 'true' this setter
     * must be able to accept an instance of 'File' as the bundle will inject one here
     * during Doctrine hydration.
     *
     * @param File|null $imageFile
     */
    public function setImageFile(?File $imageFile = null): void
    {
        $this->imageFile = $imageFile;

        if (null !== $imageFile) {
            // It is required that at least one field changes if you are using doctrine
            // otherwise the event listeners won't be called and the file is lost
            $this->updatedAt = new \DateTimeImmutable();
        }
    }

    public function getImageFile(): ?File
    {
        return $this->imageFile;
    }

    public function setImageName(?string $imageName): void
    {
        $this->imageName = $imageName;
    }

    public function getImageName(): ?string
    {
        return $this->imageName;
    }

    public function setImageSize(?int $imageSize): void
    {
        $this->imageSize = $imageSize;
    }

    public function getImageSize(): ?int
    {
        return $this->imageSize;
    }

    /**
     * @return mixed
     */
    public function getId()
    {
        return $this->id;
    }
}
```

**Controller** 

```
$form=$this->createForm(ConfigType::class);
        $form->handleRequest($request);

        if ($request->isMethod('POST')) {



            if ($form->isSubmitted() && $form->isValid()) {



                $data = $form->getData();
                $mediaImage = new MediaImage();

                /** @var UploadedFile $logoFile */
                $logoFile = $data['logo'];

                $mediaImage->setImageFile($logoFile);



                $entityManager = $this->getDoctrine()->getManager();
                $entityManager->persist($mediaImage);
                $entityManager->flush();
                $this->addFlash(
                    'notice',
                    'Your changes were saved!'
                );


            }
        }



        return $this->render('admin/config/edit.html.twig',[
            'config'=> $configArray,
            'form'=>$form->createView()

        ]);
```

**My Form**

` ->add('title',TextType::class)
            ->add('logo',VichImageType::class)`

**Comment by garak** [comment]
How your form is supposed to be "custom"?

**Comment by garak** [comment]
Closing for missing feedback

**Comment by mellamy** [comment]
Hi, sorry to dig up such an old issue but it's the only one I could find with the exact same error message as the one I got.

Just so if anyone ever ends up here from Google like I did, not knowing why this error happened, in my case it was an issue with the php.ini configuration not allowing uploads over 2Mb in size. The PHP error was silent and it gave an array instead of an object to VichUploader as far as I understand.

So check the value of `upload_max_filesize` in your php.ini.
